### PR TITLE
feat: update report function to report multiple objects

### LINKED ISSUE #21: Handle rules that generate violations that relate to more than one node
An example of this are the `*-prefer-shared` style rules. These will generate an individual violation for every layer with matching layer style. Leading to feedback like:

Screenshot 2020-04-14 at 09 54 43

Proposed solution:

Update violations so they reference an array of nodes, no longer just a single node.

This will require changes to the types, utils and maybe assistants too (unless the change can be made in a non-breaking fashion).

Once the changes are made we should update Sketch to support the new data structures.

### PR DESCRIPTION
**Pull request by jedrichards**
Resolves #21

### CODE REVIEW

File: assistants/core/src/index.ts [hunk 1]
```diff
@@ -48,7 +48,7 @@ import zhHansMessages from './locale/zh-Hans/messages'
 
 export type CreateRuleFunction = (i18n: I18n) => RuleDefinition
 
-const SUPPORTED_LOCALES = ['en', 'zh-Hans']
+const SUPPORTED_LOCALES = ['en', 'zh-Hans', 'zh-CN']
```
Comment (christianklotz) [inline assistants/core/src/index.ts:51]: I wonder why _Chinese (China)_ got added? Sketch only supports _English_ and _Chinese (Simplified)_, i.e. `zh-Hans`. However I know that Crowdin by default uses `zh-CN` and therefor the project custom language setting, see [their documentation](https://support.crowdin.com/changing-target-languages/#custom-languages).

Comment (jedrichards) [reply assistants/core/src/index.ts:51]: Ah I just added this while testing the localisation in passing. Should have been discussed in a separate issue.

If you set your region as _Chinese Simplified_ in macOS, then the locale reported by the OS (`defaults read -globalDomain AppleLocale`) is `zh-CN`, not `zh-Hans`. As I understand it `zh-Hans` is the more modern/correct value, but many systems still use `zh-CN`.

So this isn't technically adding `zh-CN` as new locale to localise for (that is defined by the po files), rather it's supporting `zh-CN` as an alias for `zh-Hans`:

```
catalogs: {
  en: enMessages,
  'zh-Hans': zhHansMessages,
  'zh-CN': zhHansMessages,
},
```

Can remove if you like.

**Comment by jedrichards** [comment]
This change is shelved until we've explored the design and native implications, and additionally explored how it relates to violation ignoring.

**Comment by jedrichards** [comment]
Parent branch has diverged sufficiently such that this branch and PR is going to be a pain to merge. I'm going to re-make it and cherry pick what I need.

**Pull request closed without merging**